src/core/service.c

```diff
@@ -2758,6 +2761,59 @@ static void service_run_next_main(Service *s) {
         (void) service_set_main_pidref(s, TAKE_PIDREF(pidref), &s->main_command->exec_status.start_timestamp);
 }
 
+static bool service_should_reload_extensions(Unit *u) {
+        Service *s = ASSERT_PTR(SERVICE(u));
+        int r;
+
+        /* Only support this for notify-reload service types. */
+        if (s->type != SERVICE_NOTIFY_RELOAD)
+                return false;
+
+        /* TODO: Add support for user services, which can use
+         * ExtensionDirectories= + notify-reload. For now, skip for user
+         * services. */
+        if (u->manager->runtime_scope != RUNTIME_SCOPE_SYSTEM) {
+                log_unit_debug(UNIT(s), "Not reloading extensions for user services.");
+                return false;
+        }
+
+        r = exec_context_has_vpicked_extensions(&s->exec_context);
```

Comment:
hmm, i figure people will be surprised by this. this requires explicit documentation. i am fine with this restriction though.

(most folks would probably expect us to look at mtime instead, but given we want to push people towards vpick I think it makes sense to restrict things to this)

Reply:
ah, i see you did document this in a later commit. good then.